napari/layers/_shapes_layer/model.py

```diff
@@ -0,0 +1,1568 @@
+
+    @property
+    def data(self):
+        """ShapeList: object containing all the shape data
+        """
+        return self._data
+
+    @data.setter
+    def data(self, data):
+        self._data = data
+        self.refresh()
+
+    @property
+    def edge_width(self):
+        """float: width of edges in px
+        """
+        return self._edge_width
+
+    @edge_width.setter
+    def edge_width(self, edge_width):
+        self._edge_width = edge_width
+        if self._update_properties is True:
+            index = self.selected_shapes
+            for i in index:
+                self.data.update_edge_width(i, edge_width)
+            self.refresh()
+        self.events.edge_width()
+
+    @property
+    def edge_color(self):
+        """str: color of edges and lines
+        """
+        return self._edge_color
+
+    @edge_color.setter
+    def edge_color(self, edge_color):
+        self._edge_color = edge_color
+        if self._update_properties is True:
+            index = self.selected_shapes
+            for i in index:
+                self.data.update_edge_color(i, edge_color)
+            self.refresh()
+        self.events.edge_color()
+
+    @property
+    def face_color(self):
+        """str: color of faces
+        """
+        return self._face_color
+
+    @face_color.setter
+    def face_color(self, face_color):
+        self._face_color = face_color
+        if self._update_properties is True:
+            index = self.selected_shapes
+            for i in index:
+                self.data.update_face_color(i, face_color)
+            self.refresh()
+        self.events.face_color()
+
+    @property
+    def opacity(self):
+        """float: Opacity value between 0.0 and 1.0.
+        """
+        return self._opacity
+
+    @opacity.setter
+    def opacity(self, opacity):
+        if not 0.0 <= opacity <= 1.0:
+            raise ValueError('opacity must be between 0.0 and 1.0; '
+                             f'got {opacity}')
+
+        self._opacity = opacity
+        if self._update_properties is True:
+            index = self.selected_shapes
+            for i in index:
+                self.data.update_opacity(i, opacity)
+            self.refresh()
+        self.events.opacity()
+
+    @property
+    def selected_shapes(self):
+        """list: list of currently selected shapes
+        """
+        return self._selected_shapes
+
+    @selected_shapes.setter
+    def selected_shapes(self, selected_shapes):
+        self._selected_shapes = selected_shapes
+        self._selected_box = self.interaction_box(selected_shapes)
+
+        # Update properties based on selected shapes
+        face_colors = list(set([self.data.shapes[i]._face_color_name for i in
+                           selected_shapes]))
+        if len(face_colors) == 1:
+            face_color = face_colors[0]
+            with self.block_update_properties():
+                self.face_color = face_color
```

Comment:
I think this will be tricky as the `face_color` property can also get set from the GUI and is then used to set the color of the next drawn shape, so I can't only have a `get_face_color` method on the shapeslist